### DISCUSSION TITLE
DataTable里的renderExpand函数目前能否支持异步？

### DISCUSSION THREAD

**Discussion by guopenghui** [tusen-ai/naive-ui#4125] [Q&A] [answered]
我看到现在那个函数的签名是返回一个vnode，是同步的。不知道现在有没有能异步来提供这个vnode的方法？

比如对于某一行，有一些信息不方便在行里面展示，所以就放在展开里面了。这些展开信息可能数据量较大，或者用户点开查看的概率较低，不值得一次性都从后端拿过来，想要在用户展开某一行的时候再去请求这一行的数据。

**Comment by guopenghui** [comment, marked as answer]
啊，我发现可以用异步组件

**Reply by gn00678465** [reply]
可以請問你怎麼實現這個功能嗎?

**Reply by guopenghui** [reply]
把要显示的内容写成异步组件，然后在函数里返回这个异步组件。

可以参考这里：

https://github.com/guopenghui/obsidian-language-learner/blob/d5928f0b6cf922d702406b2269f85e3b3734aedd/src/views/DataPanel.vue#L148-L152
